vec3.h

```diff
@@ -0,0 +1,120 @@
+//code by Ameya Vikrama Singh
+#include<iostream>
+#include<cmath>
+class vec3{
+    public:
+        float x,y,z;
+
+        vec3(){
+            x=y=z=0;
+        }
+
+        //Cartesian coordinate Constructor
+        vec3(float x,float y,float z){
```

Comment:
If you use member initializer lists, then you don't need to use the `this` pointer.